samples/manageSnapshots.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import vm_automation
+
+def takeSnapshot(vm, snapshotName, args):
+    wasPoweredOn = vm.isPoweredOn()
+    if args.powerOn:
+        vm.powerOn()
+    if args.powerOff:
+        vm.powerOff
+    retVal = vm.takeSnapshot(args.snapshotName)
+    if wasPoweredOn:
+        vm.powerOn()
+    else:
+        vm.powerOff()
+    return retVal
+
+def deleteSnapshots(vm, snapshotName):
+    vm.getSnapshots()
+    for snapshot in vm.snapshotList:
+        if snapshot[0].name == snapshotName:
+            vm.deleteSnapshot(snapshotName)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-k", "--keyword", help="VM search parameter")
+    parser.add_argument("-sn", "--snapshotName", help="snapshot name")
+    parser.add_argument("-a", "--action", help="action [create|revert|delete]")
+    parser.add_argument("-hf", "--hypervisorConfig", help="hypervisor config file")
+    parser.add_argument("-p1", "--powerOn", help="power on vm before snapshot", action="store_true")
+    parser.add_argument("-p0", "--powerOff", help="power off vm before snapshot", action="store_true")
```

Comment:
Removing `-p0` && `-p1` as they are not used here.

Reply:
Well I am incorrect this is used, but should it be?  Snapshot can occur with power on or off just a diff in storage used.

Reply:
Won't hurt to leave for now, can change minds later, landing...